### ISSUE TITLE
Detect if WP Rocket is on one.com environment

### ISSUE THREAD

**Issue by sebssingh** [wp-media/wp-rocket#5235]
Hi
Similar to what WP Rocket has for other hosting providers such as seen in:
https://github.com/wp-media/wp-rocket/blob/develop/inc/ThirdParty/Hostings/HostResolver.php

It would be great if this could also be done for domains on one.com environment.

We have a lot of server variables for instance $_SERVER['ONECOM_DOMAIN_NAME'] that would only be present on the one.com environment.

So if you have a domain let's call it domain.tld the ONECOM_DOMAIN_NAME server variable would be "domain.tld", simply detecting if this variable gives any value would be good enough to detect if it is on a one.com environment or not.

The reason for this request is that once this detection is in place, we can also improve some other things such as certain tailor-made pre-selected settings for domains on one.com environment but for those I will create separate tickets as to not make this one too cluttered.

Thank you and please let me know if any further information is required from my side.

**Comment by jeawhanlee** [comment]
### Scope a solution ✅
We will update WP_Rocket\ThirdParty\Hostings::HostResolver to check for one.com env
```
if ( isset( $_SERVER['ONECOM_DOMAIN_NAME'] ) ) {
	self::$hostname = 'onecom';
	return 'onecom';
}
```
Then we will update `WP_Rocket\ThirdParty\Hostings::HostSubscriberFactory` .

Related tests should also take most effort.

Estimate the effort ✅
[S]

**Comment by remyperona** [comment]
@piotrbak shouldn't this one be closed in favour of #5189?

**Comment by piotrbak** [comment]
@Tabrisrp Yes, it can be easily closed now. Thanks!
